### PR TITLE
UI API Redux updates

### PR DESCRIPTION
**Pull request by cdcabrera**
Return promises with Redux middleware. Clean up action boilerplate
  - fix post and put submissions
  - updates credentials edit/new dialog

updates #534 

@jeff-phillips-18  ~~looks like the property for passing ```credentialType``` formerly known as ```createCredentialType``` is acting up, conflict between being edited vs new. Also need to amend the commits.~~ Solved

### CODE REVIEW

File: client/src/components/credentials/credentials.js [hunk 1]
```diff
@@ -290,10 +290,12 @@ class Credentials extends React.Component {
   }
 
   render() {
-    const { loading, loadError, errorMessage, credentials } = this.props;
+    // const { loading, loadError, errorMessage, credentials } = this.props;
```
Comment (jeff-phillips-18) [inline client/src/components/credentials/credentials.js]: OK to remove this?

File: client/src/components/credentials/credentials.js [hunk 2]
```diff
@@ -290,10 +290,12 @@ class Credentials extends React.Component {
...
+    const { getPending, getError, getErrorMessage, credentials } = this.props;
     const { filteredItems, selectedItems } = this.state;
 
-    if (loading) {
+    // if (loading) {
```
Comment (jeff-phillips-18) [inline client/src/components/credentials/credentials.js]: OK to remove this?

File: client/src/components/credentials/credentials.js [hunk 3]
```diff
@@ -303,11 +305,13 @@ class Credentials extends React.Component {
         </Modal>
       );
     }
-    if (loadError) {
+
+    // if (loadError) {
```
Comment (jeff-phillips-18) [inline client/src/components/credentials/credentials.js]: OK to remove this?

File: client/src/components/credentials/credentials.js [hunk 4]
```diff
@@ -352,33 +356,27 @@ class Credentials extends React.Component {
...
+  getErrorMessage: PropTypes.string,
+
   activeFilters: PropTypes.array,
   sortType: PropTypes.object,
   sortAscending: PropTypes.bool
 };
 
 Credentials.defaultProps = {
-  loading: true
+  // loading: true
```
Comment (jeff-phillips-18) [inline client/src/components/credentials/credentials.js]: Remove commented out code

File: client/src/routes.js [hunk 5]
```diff
@@ -13,19 +13,25 @@ const routes = () => [
     icon: 'pficon pficon-network',
     title: 'Sources',
     to: '/sources',
+    // href: '/sources',
+    // authenticate: true,
```
Comment (jeff-phillips-18) [inline client/src/routes.js]: remove

File: client/src/routes.js [hunk 6]
```diff
@@ -13,19 +13,25 @@ const routes = () => [
...
     redirect: true,
     component: Sources
   },
   {
     icon: 'fa fa-wifi',
     title: 'Scans',
     to: '/scans',
+    // href: '/scans',
+    // authenticate: true,
```
Comment (jeff-phillips-18) [inline client/src/routes.js]: remove

File: client/src/components/credentials/credentials.js (hunk 1 above)
Comment (cdcabrera) [reply client/src/components/credentials/credentials.js]: whoops, looks like cleanup

**Pull request merged**